test/pages/ReporterTypePageSpec.scala

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package pages
+
+import models.ReporterType._
+import models.matching.RegistrationInfo
+import models._
+import org.scalacheck.Arbitrary.arbitrary
+import pages.behaviours.PageBehaviours
+import uk.gov.hmrc.domain.Nino
+import org.scalatest.matchers.should.Matchers._
+
+import java.time.LocalDate
+
+class ReporterTypePageSpec extends PageBehaviours {
+
+  private val testParamGenerator = for {
+    addressLookup    <- arbitrary[models.AddressLookup]
+    address          <- arbitrary[models.Address]
+    postcode         <- arbitrary[String]
+    name             <- arbitrary[models.Name]
+    booleanField     <- arbitrary[Boolean]
+    nino             <- arbitrary[Nino]
+    registrationInfo <- arbitrary[RegistrationInfo]
+    dob              <- arbitrary[LocalDate]
+    stringField      <- arbitrary[String]
+    utr              <- arbitrary[UniqueTaxpayerReference]
+  } yield (addressLookup, address, postcode, name, booleanField, nino, registrationInfo, dob, stringField, utr)
+
+  "ReporterTypePage" - {
+
+    beRetrievable[ReporterType](ReporterTypePage)
+
+    beSettable[ReporterType](ReporterTypePage)
+
+    beRemovable[ReporterType](ReporterTypePage)
+
+    "cleanUp" - {
+      "must not clear answers" - {
+        "when answer changes to 'Sole Trader'" in {
+          // spot check
+          forAll(testParamGenerator) {
+            case (addressLookup, address, postcode, name, booleanField, nino, registrationInfo, dob, stringField, utr) =>
+              val ua = emptyUserAnswers
+                .withPage(IndWhatIsYourNINumberPage, nino)
+                .withPage(IndContactNamePage, name)
+                .withPage(IndDateOfBirthPage, dob)
+                .withPage(RegistrationInfoPage, registrationInfo)
+                .withPage(IndWhatIsYourNamePage, name)
+                .withPage(DateOfBirthWithoutIdPage, dob)
+                .withPage(IndWhereDoYouLivePage, booleanField)
+                .withPage(IndWhatIsYourPostcodePage, postcode)
+                .withPage(AddressLookupPage, Seq(addressLookup))
+
+              val result = ReporterTypePage.cleanup(Some(LimitedCompany), ua).success.value
+
+              result.get(IndWhatIsYourNINumberPage) should (be(None) or not be empty)
```

Comment:
I don't think any of these will be `None`? Can they all just be `should not be empty`?